lib/matplotlib/text.py

```diff
@@ -228,7 +230,18 @@ def _get_multialignment(self):
 
     def get_rotation(self):
         """Return the text angle in degrees between 0 and 360."""
-        return get_rotation(self._rotation)  # string_or_number -> number
+        if self.get_transform_rotates_text():
+            angle = get_rotation(self._rotation)
+            x, y = self.get_unitless_position()
+            return self.get_transform().transform_angles([angle, ], [[x, y]])
```

Comment:
This seems like it has type instability?

Reply:
@tacaswell resolved, please have a look